app/services/change_schedule.rb

```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+
+class ChangeSchedule
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :cpd_lead_provider
+  attribute :participant_id
+  attribute :reason
+  attribute :course_identifier
+  attribute :schedule_identifier
+  attribute :cohort
+
+  alias_attribute :user_profile, :participant_profile
+  delegate :participant_profile_state, to: :participant_profile, allow_nil: true
+  delegate :lead_provider, to: :cpd_lead_provider, allow_nil: true
+
+  validates :course_identifier, course: true, presence: { message: I18n.t(:missing_course_identifier) }
+  validates :participant_id, presence: { message: I18n.t(:missing_participant_id) }
+  validates :participant_id, format: { with: /\A[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}\Z/, message: I18n.t("errors.participant_id.invalid") }, allow_blank: true
+  validates :cpd_lead_provider, presence: { message: I18n.t(:missing_cpd_lead_provider) }
+  validate :participant_has_user_profile
```

Comment:
can we use this validation instead as it will validate the participant profile and induction record, as well as the cpd lead provider is linked to the profile?`validates :cpd_lead_provider, induction_record: true`

Reply:
For some reason some pre-existing tests started to fail after I added the `induction_record: true` validation. So kept how it was in the first place.